consensus/consensus-types/src/block_retrieval.rs

```diff
@@ -15,17 +15,26 @@ pub const NUM_PEERS_PER_RETRY: usize = 3;
 pub const RETRY_INTERVAL_MSEC: u64 = 500;
 pub const RPC_TIMEOUT_MSEC: u64 = 5000;
 
+/// TODO double check backwards compatability was done correctly here and
+/// on `ConsensusMsg`
+#[derive(Serialize, Deserialize, Clone, Debug, PartialEq, Eq)]
+pub enum BlockRetrievalRequest {
```

Comment:
TODO need to figure out how to make this work.
- `BlockRetrievalRequestV1` makes a request for all of the blocks between two hash values (i.e., `commit_root_hash` and some other hash).
- `BlockRetrievalRequestV2` makes a request for all of the blocks between `commit_root_hash` and some round where the round can be the `window_root`. But if the window_root is behind the `commit_root`, there's not a guarantee that a node will have that history... cc @bchocho to double check if this is is accurate